server/src/main/java/io/crate/execution/jobs/JobSetup.java

```diff
@@ -637,6 +639,9 @@ public Void visitPKLookup(PKLookupPhase pkLookupPhase, Context context) {
         @Override
         public Void visitMergePhase(final MergePhase phase, final Context context) {
             boolean upstreamOnSameNode = context.opCtx.upstreamsAreOnSameNode(phase.phaseId());
+            if (upstreamOnSameNode) {
+                throw new IllegalStateException("Same node");
+            }
```

Comment:
this was never thrown neither on master, nor on [approach 1](https://github.com/crate/crate/pull/13718/commits/ca6679135778b14d8f874894eecf9fc622783941) (always do merge), nor on [approach 2](https://github.com/crate/crate/pull/13718/commits/b1de041aff76d38888e7da21d5726f261557aa47) (switch to SAME_NODE)